src/main.rs

```diff
@@ -51,6 +53,21 @@ fn read_file(filename: &str) -> std::result::Result<String, String> {
     read_to_string(filename).map_err(|e| format!("Failed to read {filename}: {e}"))
 }
 
+fn wait_for_device(root_device: &str) -> Result<()> {
+    let mut counter = 0;
+    while !Path::new(&root_device).exists() {
+        let duration = time::Duration::from_millis(5);
+
+        if counter > 1000 {
+            return Err("timout reached while waiting for the device".into());
+        }
+
+        thread::sleep(duration);
+        counter += 1;
+    }
+    Ok(())
+}
```

Comment:
The `main.rs` already contains a few helper functions like this one, so adding this one there is consistent.
But I think they should be moved somewhere else in the future (`utils.rs`?) to de-clutter the `main.rs` (it is the entrypoint for newcomers to a project and should thus, in my opinion, be the easiest file to read).

Reply:
Good point. We have #6 do move stuff into a separate file. But that PR tried to do too much stuff at once and has not been merged so far.